### PR TITLE
Added IBlockSearchableText to enable extraction of text from any block type

### PR DESCRIPTION
**Pull request by tiberiuichim**

### CODE REVIEW

File: docs/source/blocks.rst [hunk 1]
```diff
@@ -76,3 +76,31 @@ Storing blocks is done also via a default PATCH content operation::
...
+
+SearchableText indexing for blocks
+----------------------------------
+
+As the main consumer of plone.restapi's blocks, this functionality is specific
+to Volto blocks implementation. By default searchable text (for Plone's
+SearchableText index) is extracted from `text` blocks.
+
+To extract searchable text for other types of blocks, you need to write an
+adaptor that can process that type of block.::
```
Comment (sneridagh) [inline docs/source/blocks.rst:88]: Typo

**Comment by sneridagh** [comment]
@jenkins-plone-org please run jobs

**Comment by tiberiuichim** [comment]
@sneridagh I couldn't see the test failure in the  pull-request-5.2-3.6 job. Maybe you know what's that about?

**Comment by sneridagh** [comment]
Nope, let's run it again :(

**Comment by sneridagh** [comment]
@jenkins-plone-org please run jobs

**Comment by tiberiuichim** [comment]
@jenkins-plone-org please run jobs

**Comment by jensens** [comment]
Now we are at a point where this is all far away from an Restapi use case. This all need to be moved to plone.app.contenttypes or plone.app.dexterity, or wherever it better fits in (please no new package, we have already enough).

**Comment by tisto** [comment]
This is a refactoring of the indexing functionality that already existed in plone.restapi. We have too keep things together for now, otherwise, it will become impossible to maintain plone.restapi as a package that is used both in Plone core as well as an add-on product. We certainly have to discuss the package architecture of this in the future. Though, as long as Volto is not merged in or shipped with Plone 6, we just can not start to spread functionality across different packages.

**Comment by tisto** [comment]
@tiberiuichim can you please fix the merge conflicts here?

**Comment by tiberiuichim** [comment]
@jenkins-plone-org please run jobs

**Comment by tiberiuichim** [comment]
@jenkins-plone-org please run jobs

**Pull request merged**